docs/README.md

```diff
@@ -99,3 +99,55 @@ $ git commit
 Create a pull request once you are ready to submit your changes.
 
 We will review your changes, add any feedback and once everything looks good merge your changes into the mainline.
+
+
+## Advanced Usage
+
+### Re-generating diagrams for API docs
+
+Generate new grammars and diagrams when updating the docs as follows:
+
+_Note: Modifications will typically be in the `docs/content/latest/api/` either `ysql` or `ycql` subfolder.
+Will assume that the subfolder is `docs/content/latest/api/ysql` and the target file is `commands/cmd_copy.md` in the examples below_
+
+1. Update (edit) the appropriate `ebnf` source grammar file with your changes (e.g. adding a new supported statement or clause).
+_Example: for the YSQL API that is the `./content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf` file._
+
+2. If you are adding a new file (e.g. for a new statement), use the template `includeMarkdown` macro (see `cmd_copy.md` file mentioned below as an example).
+
+_Example: for the YSQL `COPY` command the source file is `./content/latest/api/ysql/commands/cmd_copy.md`._
+
+3. Create two (empty) files `<name>.grammar.md` and `<name>.gramar.md` inside the `syntax_resources` folder.
+The `<name>` should be a comma-separated list of rule names from the `ebnf` file that you want in your new file.
+Also, the files location should match the directory structure from the top-level `ysql` (or `ycql`) folder
```

Comment:
The two new files must be added into a directory structure that matches the top-level directory (`ysql` or `ycql`) — if needed, create any required parent directories.